### PR TITLE
Next.js: Move sharp into optional deps

### LINKED ISSUE #25075: [Bug]: Dependency on sharp in @storybook/nextjs should be optional
### Describe the bug

As of [7.6.0](https://github.com/storybookjs/storybook/compare/v7.5.3...v7.6.0) @nextjs/storybook has a direct dependency on [sharp](https://sharp.pixelplumbing.com/install), in order to handle AVIF processing. This is an optional dependency in nextjs itself so presents an issue for anyone (like us) who are running Storybook inside of a tightly constrainted CI environment - we have no need for AVIF or Next.js's image optimisation, so we don't need the sharp dependency and don't want to whitelist the download domains in order for it to download binaries at install time.

It would be better for this to be an optional dependency, and to throw an error if an AVIF file is detected - this would match Next's own behaviour (https://nextjs.org/docs/messages/install-sharp)

### To Reproduce

_No response_

### System

```bash
Storybook Environment Info:

  System:
    OS: macOS 14.1.1
    CPU: (10) arm64 Apple M1 Max
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.10.0 - ~/.asdf/installs/nodejs/20.10.0/bin/node
    Yarn: 1.22.19 - ~/.asdf/shims/yarn
    npm: 10.2.4 - ~/.asdf/plugins/nodejs/shims/npm <----- active
  Browsers:
    Chrome: 119.0.6045.199
    Edge: 119.0.2151.97
    Safari: 17.1
  npmPackages:
    @storybook/addon-actions: 7.6.3 => 7.6.3
    @storybook/addon-essentials: 7.6.3 => 7.6.3
    @storybook/addon-interactions: 7.6.3 => 7.6.3
    @storybook/addon-links: 7.6.3 => 7.6.3
    @storybook/blocks: 7.6.3 => 7.6.3
    @storybook/nextjs: 7.6.3 => 7.6.3
    @storybook/react: 7.6.3 => 7.6.3
    @storybook/testing-library: 0.2.2 => 0.2.2
    chromatic: 9.1.0 => 9.1.0
    eslint-plugin-storybook: 0.6.15 => 0.6.15
    storybook: 7.6.3 => 7.6.3
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by shuta13**
## What I did

Fixed https://github.com/storybookjs/storybook/issues/25075

I referred [implementations of the nextjs image optimizer](https://github.com/vercel/next.js/blob/167ea3382fe6b113dbcf4272d55ce04ac314c51c/packages/next/src/server/image-optimizer.ts) for moving `sharp` into optional dependency.

#### The changes in this PR are covered in the following automated tests:
- [x] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

1. Run `yarn task --task serve --start-from auto --template nextjs/13-ts serve`
2. Open Storybook in your browser
3. Access Avif Story( The avif image is shown correctly. )

### Documentation

- [x] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

### CODE REVIEW

File: code/frameworks/nextjs/src/next-image-loader-stub.ts [hunk 1]
```diff
@@ -37,10 +45,16 @@ const nextImageLoaderStub: RawLoaderDefinition<LoaderOptions> = async function N
...
-    height = result.height;
+    if (sharp) {
+      const transformer = sharp(content);
+      const result = await transformer.metadata();
+      width = result.width;
+      height = result.height;
+    } else {
+      throw new Error(
+        'You have to install sharp in order to use image optimization features in Next.js.'
+      );
```
Comment (shuta13) [inline code/frameworks/nextjs/src/next-image-loader-stub.ts:56]: I wanted to add the part of the documentation for resolving this error like [nextjs messages docs](https://nextjs.org/docs/messages/install-sharp) but I could not find the part to modify in storybook docs. Could you tell me if it exists.

Comment (valentinpalkovic) [reply code/frameworks/nextjs/src/next-image-loader-stub.ts:56]: You could describe the error in the faq section [here](https://github.com/storybookjs/storybook/blob/next/docs/get-started/nextjs.md#faq)

**Comment by valentinpalkovic** [comment]
Hi @shuta13 

Thank you so much for your contribution! :)

I have edited the `throw new Error` statement actually to use Storybook's error framework.

File: code/frameworks/nextjs/src/next-image-loader-stub.ts (hunk 1 above)
Comment (shuta13) [reply code/frameworks/nextjs/src/next-image-loader-stub.ts]: I added the part for resolving the error in the FAQ section you have shown. Please check out: https://github.com/storybookjs/storybook/pull/26787/commits/3eaacff2aeef09fb38fc12d3d886b50d8ee18cab

**Review by valentinpalkovic — APPROVED** [review]

**Pull request merged**